feat: allow to connect a python rpc-client to an rpc-server via a unix FIFO file instead of subprocess

hpk42 opened the pull request:
also internally refactors rpc.py a bit and strikes unneccessary `**kwargs`

link2xt commented on deltachat-rpc-client/tests/test_rpc_fifo.py:
```diff
@@ -0,0 +1,22 @@
+    fn_response_fifo = tmp_path.joinpath("response_fifo")
+    os.mkfifo(fn_request_fifo)
+    os.mkfifo(fn_response_fifo)
+    popen = subprocess.Popen(f"deltachat-rpc-server <{fn_request_fifo} >{fn_response_fifo}", shell=True)
```
For any real use FIFOs should likely be replaced with the UNIX socket, which exists on all platforms (including Windows) and can handle multiple connections.

link2xt commented on deltachat-rpc-client/src/deltachat_rpc_client/_utils.py:
```diff
@@ -47,36 +47,33 @@ def __setattr__(self, attr, val):
     """
     from .client import Bot
 
-    _run_cli(Bot, hooks, argv, **kwargs)
```
Don't know if it is actually unused e.g. for `rpc_server_path`, at least the documentation should be updated.

adbenitez replied:
the reason `**kwargs` existed in the past was to be able to control the created process, ex. IIRC TUI clients need to tweak it so stderr don't dirty the screen,

I am talking about the removal in the Rpc class, but other approaches like allowing to pass a Popen directly to the start method in RpcProcess would also work I guess

Simon-Laux reviewed (approved):
Looks good to me.

- Maybe removal of `**kwargs` should be mentioned in changelog as API change.
- we should wait for another approval from someone that knows python better than I do.

hpk42 committed:
feat: allow to connect a python rpc-client to an rpc-server via a unix FIFO file instead of subprocess

Splits the RPC client into a transport-agnostic BaseRpc plus RpcSubprocess
(aliased as Rpc for backwards compatibility) and the new RpcFIFO.
Also strikes unneccessary `**kwargs`.
